tools/flake8-torch/flake8_torch/checker.py

```diff
@@ -0,0 +1,96 @@
+from pathlib import Path
+import ast
+import yaml
+
+
+class TorchVisitor(ast.NodeVisitor):
+    def __init__(self):
+        # Map from import alias (may be the same as name) to import name
+        self._import_info = {}
+        self._call_info = []
+
+    def visit_Call(self, node):
+        func_name = ""
+        curr = node.func
+        while isinstance(curr, ast.Attribute):
+            func_name = "." + curr.attr + func_name
+            curr = curr.value
+        if isinstance(curr, ast.Name):
+            func_name = curr.id + func_name
+
+        self._call_info.append(
+            {"name": func_name, "lineno": node.lineno, "col_offset": node.col_offset}
+        )
+
+        self.generic_visit(node)
+
+    def visit_Import(self, node):
+        for alias in node.names:
+            if alias.asname is None:
+                self._import_info[alias.name] = alias.name
+            else:
+                self._import_info[alias.asname] = alias.name
+
+        self.generic_visit(node)
+
+    def visit_ImportFrom(self, node):
+        module = "" if node.module is None else node.module
+        module = "." * node.level + module
+        for alias in node.names:
+            if alias.asname is None:
+                self._import_info[alias.name] = module + "." + alias.name
+            else:
+                self._import_info[alias.asname] = module + "." + alias.name
+        self.generic_visit(node)
+
+
+class TorchChecker:
+    name = "flake8-torch"
+    version = "0.0.1"
+
+    def __init__(self, tree):
+        self.tree = tree
+        self.deprecated_config = {}
+        with open(Path(__file__).absolute().parent / "deprecated_symbols.yaml") as f:
+            for item in yaml.load(f, yaml.SafeLoader):
+                self.deprecated_config[item["name"]] = item
```

Comment:
this could be a good place to check for duplicate items with the same name

Reply:
Maybe later.